qa/rpc-tests/mempool_size_limit_more.py

```diff
@@ -0,0 +1,298 @@
+            if len(txs) > 0:
+                tx = txs.pop(0)
+            else:
+                assert(len(self.utxos) > 0)
+                tx = self.create_a_big_transaction(self.utxos)
+            if tx['feerate'] <= Decimal('0.00015'):
+                discarded_txs += 1
+                continue
+            tx_hex = tx['hex']
+            feerate = tx['feerate']
+            self.nodes[0].sendrawtransaction(tx_hex, True)
+            last_tx_size = len(tx_hex)//2
+            minfee, maxfee = min(minfee, feerate), max(maxfee, feerate)
+            tx_sent += last_tx_size
+            usage = self.nodes[0].getmempoolinfo()
+
+        mark_logs(cc('g', "Rejected transactions (fee too low for this test): ") + f"{discarded_txs}", self.nodes, DEBUG_MODE)
+        mark_logs(cc('g', "transactions -- maxfee: ") + f"{maxfee:1.8f}" + cc('g', " - minfee: ") + f"{minfee:1.8f}", self.nodes, DEBUG_MODE)
+        mempoolstatus = self.nodes[0].getmempoolinfo()
+        mark_logs(cc('g', "Current mempool state: bytes: ") + f"{mempoolstatus['bytes']}" +
+                  cc('g', " - bytes-for-tx: ") + f"{mempoolstatus['bytes-for-tx']}" +
+                  cc('g', " - bytes-for-cert: ") + f"{mempoolstatus['bytes-for-cert']}", self.nodes, DEBUG_MODE)
+        # Assert that we did not evict any certificate
+        assert_equal(mempoolstatus['bytes-for-cert'], certPartitionStatus)
+        print()
+
+
+        # We choose a feerate that still classify as high: the midpoint between the lower and higher tx fee of the preious step
+        selected_high_feerate = minfee + (maxfee - minfee) / 2
+        mark_logs(f"Filling tx partition with tiny transactions having feerate {selected_high_feerate:1.8f} (high)", self.nodes, DEBUG_MODE, color = 'c')
+        mark_logs("It's ok to fill tx partition over its limit, we only do not want to evict certificates", self.nodes, DEBUG_MODE, color = 'c')
+        last_tx_size = 0
+        usage = self.nodes[0].getmempoolinfo()
+        addr = self.nodes[0].getnewaddress()
+        utxos = self.nodes[0].listunspent()
+        while(int(usage['bytes']) < NODE0_LIMIT_B - last_tx_size):
+            t = utxos.pop(0)
+            inputs = [{"txid" : t["txid"], "vout" : t["vout"]}]
+            outputs = {addr: t["amount"] - Decimal(str(selected_high_feerate))}
+            rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+            signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")
+            try:
+                self.nodes[0].sendrawtransaction(signedtx['hex'])
+            except JSONRPCException as e:
+                errorString = e.error['message']
+                print(errorString)
+                assert(False)
+            new_usage = self.nodes[0].getmempoolinfo()
+            last_tx_size = int(new_usage['bytes-for-tx']) - int(usage['bytes-for-tx'])
+            usage = new_usage
```

Comment:
The transaction being here created has not fee rate equal to `selected_high_feerate`, it has fee equal to `selected_high_feerate`. Since fee rate is used by acceptance/rejection logic, the transaction must be created in a different way:

```suggestion
        while(int(usage['bytes']) < NODE0_LIMIT_B - last_tx_size):
            t = utxos.pop(0)
            inputs = [{"txid" : t["txid"], "vout" : t["vout"]}]

            outputs = {addr: t["amount"]} # for initial estimation
            rawtx = self.nodes[0].createrawtransaction(inputs, outputs) # for initial estimation
            signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE") # for initial estimation
            tx_size = len(signedtx['hex'])//2 # for initial estimation

            outputs = {addr: t["amount"] - Decimal(selected_high_feerate * tx_size)}
            rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
            signedtx = self.nodes[0].signrawtransaction(rawtx, None, None, "NONE")

            try:
                txid = self.nodes[0].sendrawtransaction(signedtx['hex'], True)
                last_tx_size = len(signedtx['hex'])//2
                usage = self.nodes[0].getmempoolinfo()
            except JSONRPCException as e:
                errorString = e.error['message']
                print(errorString)
                assert(False)
```

Please note `sendrawtransaction `setting second parameter (`allowhighfees`) to `True`, as it is done also in `mempool_size_limit_more.py`. Also a minor change on the computation of `last_tx_size` is included (but you can skip it).

Reply:
Thanks! Yes, I integrated all the proposed suggestions in the three tx creation sections. Now:
* `tx_size` is evaluated beforehand
* fees are correctly evaluated from the selected feerates
* all `sendrawtransaction`s have the optional parameter set to `True` to allow over the top fees